### PR TITLE
core: default values should be set after reading flag

### PR DESCRIPTION
**Pull request by subhamkrai**
core: default values should be set after reading flag

we should read flags before setting the default values
otherwise, if someone provides a ceph cluster namespace, the
operator namespace will still point to default rook-ceph

Signed-off-by: subhamkrai <srai@redhat.com>

### CODE REVIEW

File: kubectl-rook-ceph.sh [hunk 1]
```diff
@@ -378,50 +378,50 @@ function check_mgr_pods_status_and_count() {
...
+  deployment_spec=$(jq 'del(.template.spec.containers[0].startupProbe)' <<<$deployment_spec)
+  echo "$deployment_spec" # for returning the updated deployment_spec
 }
 
-function update_deployment_spec_command(){
+function update_deployment_spec_command() {
   deployment_spec=$1
   main_command='["sleep","infinity"]'
-  deployment_spec=$(jq .template.spec.containers[0].command="$main_command" <<< $deployment_spec)
-  deployment_spec=$(jq .template.spec.containers[0].args=[] <<< $deployment_spec)
```
Comment (travisn) [inline kubectl-rook-ceph.sh:392]: The editor automatically changed all of these? Seems like this space is good to keep, unless the editor really removes it.

File: kubectl-rook-ceph.sh [hunk 2]
```diff
@@ -610,6 +601,15 @@ function parse_main_flag() {
 REMAINING_ARGS=()
 parse_flags 'parse_main_flag' "$@"
 
+# Default values
```
Comment (travisn) [inline kubectl-rook-ceph.sh:604]: We have a CI test in a namespace other that rook-ceph, right? If so, I'm curious how we missed this.

File: kubectl-rook-ceph.sh (hunk 1 above)
Comment (subhamkrai) [reply kubectl-rook-ceph.sh:392]: yes, it was the editor

File: kubectl-rook-ceph.sh (hunk 2 above)
Comment (subhamkrai) [reply kubectl-rook-ceph.sh:604]: so in the CI or generally when I did testing, we are either not passing the namespace or passing both the namespace cluster and operator. That's why it is missed in CI.

The current behavior is
1. default is rook-ceph
2.  if only the cluster namespace passed then the operator will also pick the namespace as the cluster
3. we can pass both cluster and operator namespace

The only other case I can see that I don't think is important is when the user is only passing the operator namespace, in this case, cluster namespace will rook-ceph only which I think make sense if user don't pass namespace we stick with `rook-ceph`

Comment (travisn) [reply kubectl-rook-ceph.sh:604]: Agreed on that behavior, thanks for clarifying.

**Review by travisn — APPROVED** [review]

**Pull request merged**